agenta-backend/agenta_backend/routers/organization_router.py

```diff
@@ -0,0 +1,150 @@
+else:
+    from agenta_backend.services.auth_helper import (
+        SessionContainer,
+        verify_session,
+    )
+    from agenta_backend.services.selectors import get_user_and_org_id
+
+
+router = APIRouter()
+
+
+@router.post("/add/{organization_id}/invite/")
+async def invite_to_org(
+    payload: OrganizationInvite,
+    stoken_session: SessionContainer = Depends(verify_session()),
+):
+    if os.environ["FEATURE_FLAG"] not in ["cloud", "ee", "demo"]:
+        raise HTTPException(
+            status_code=500,
+            detail="This feature is not available in the Open Source version",
+        )
+
+    try:
+        organization_id, email_address = (
+            payload.organization_id,
+            payload.email_address,
+        )
+
+        kwargs: dict = await get_user_and_org_id(stoken_session)
+        organization_access = await check_user_org_access(
+            kwargs, ObjectId(organization_id)
+        )
+
+        if organization_access:
+            organization = await get_organization((organization_id))
+            user = await engine.find_one(UserDB, UserDB.uid == kwargs["uid"])
+            if user.email == email_address:
+                return JSONResponse(
+                    {"message": "You cannot invite yourself to your own organization"},
+                    status_code=400,
+                )
+
+            token = generate_invitation_token()
+            expiration_date = datetime.utcnow() + timedelta(days=7)
+
+            send_email = send_invitation_email(email_address, token, organization, user)
+
+            if send_email:
+                created_invitation = InvitationDB(
+                    token=token,
+                    email=email_address,
+                    expiration_date=expiration_date,
+                    used=False,
+                )
+
+                organization.invitations.append(created_invitation)
+                await engine.save(organization)
+
+                return JSONResponse(
+                    {"message": "Invited user to organization"}, status_code=200
+                )
+            else:
+                return JSONResponse(
+                    {"message": "Failed to invite user to organization"},
+                    status_code=400,
+                )
+
+        else:
+            return JSONResponse(
+                {"message": "You do not have permission to access this organization"},
+                status_code=403,
+            )
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@router.post("/accept/")
+async def add_user_to_org(
+    payload: OrganizationToken,
+    background_tasks: BackgroundTasks,
+    stoken_session: SessionContainer = Depends(verify_session()),
+):
+    if not (os.environ["FEATURE_FLAG"] in ["cloud", "ee", "demo"]):
+        raise HTTPException(
+            status_code=500,
+            detail="This feature is not available in the Open Source version",
+        )
+
+    try:
```

Comment:
we should move this whole logic here to ee.services.organization_service

Reply:
I've done that.